daemon/src/taker.rs

```diff
@@ -200,72 +179,43 @@ async fn main() -> Result<()> {
                 housekeeping::rebroadcast_transactions(&mut conn, &wallet)
                     .await
                     .unwrap();
-                let cfds = load_all_cfds(&mut conn).await.unwrap();
-
-                let (cfd_feed_sender, cfd_feed_receiver) = watch::channel(cfds.clone());
-                let send_to_maker = send_to_socket::Actor::new(write)
-                    .create(None)
-                    .spawn_global();
-
-                let (monitor_actor_address, mut monitor_actor_context) = xtra::Context::new(None);
-                let (oracle_actor_address, mut oracle_actor_context) = xtra::Context::new(None);
 
-                let cfd_actor_inbox = taker_cfd::Actor::new(
-                    db.clone(),
+                let connection::Actor {
+                    send_to_maker,
+                    read_from_maker,
+                } = connection::Actor::new(opts.maker).await;
+
+                let ActorSystem {
+                    cfd_actor_addr,
+                    cfd_feed_receiver,
+                    order_feed_receiver,
+                    update_cfd_feed_receiver,
+                } = ActorSystem::new(
+                    db,
                     wallet.clone(),
                     oracle,
-                    cfd_feed_sender,
-                    order_feed_sender,
-                    update_cfd_feed_sender,
-                    Box::new(send_to_maker),
-                    monitor_actor_address.clone(),
-                    oracle_actor_address.clone(),
+                    send_to_maker,
+                    read_from_maker,
+                    |cfds, channel| oracle::Actor::new(cfds, channel),
+                    {
+                        |channel, cfds| {
+                            let electrum = opts.network.electrum().to_string();
+                            async move {
+                                monitor::Actor::new(electrum, channel, cfds.clone()).await
+                            }
```

Comment:
`async move` for a single `.await` expression should be equivalent to just passing the initial future in!

```suggestion
monitor::Actor::new(electrum, channel, cfds.clone())
```